tls_proxy.cpp fails to compile

Mac OS X 10.10.5, Xcode-7.1.1, Macports Boost-1.59.0.

Configured:
```
* This file was automatically generated Wed Dec  9 00:21:17 2015 UTC by
* uri@Uris-MacBook-Air.local running './configure.py --prefix=/opt/local --with-boost --with-bzip2 --with-lzma --with-openssl --with-zlib --with-sqlite3 --with-python-version=2.7 --with-sphinx --cc=clang --cc-abi-flags=-maes -mpclmul -mssse3 -msse4.2 -mrdrnd -I/opt/local/include'
*
* Target
*  - Compiler: clang++  -m64 -pthread -maes -mpclmul -mssse3 -msse4.2 -mrdrnd -I/opt/local/include -std=c++11 -D_REENTRANT -fstack-protector -O3
*  - Arch: x86_64/x86_64
*  - OS: darwin
```

Compilation error:
```
clang++  -m64 -pthread -maes -mpclmul -mssse3 -msse4.2 -mrdrnd -I/opt/local/include -std=c++11 -D_REENTRANT -fstack-protector -O3 -Wall -Wextra -Wstrict-aliasing -Wstrict-overflow=5 -Wcast-align -Wmissing-declarations -Wpointer-arith -Wcast-qual -Wunreachable-code -Ibuild/include -c ./src/cmd/tls_proxy.cpp -o build/obj/app/tls_proxy.o
In file included from ./src/cmd/tls_proxy.cpp:23:
In file included from build/include/botan/tls_server.h:11:
In file included from build/include/botan/tls_channel.h:14:
build/include/botan/tls_session_manager.h:130:14: warning: 'remove_all' overrides a member function
      but is not marked 'override' [-Winconsistent-missing-override]
      size_t remove_all();
             ^
build/include/botan/tls_session_manager.h:60:22: note: overridden virtual function is here
      virtual size_t remove_all() = 0;
                     ^
./src/cmd/tls_proxy.cpp:45:35: warning: unused parameter 'where' [-Wunused-parameter]
inline void log_error(const char* where, const boost::system::error_code& error)
                                  ^
./src/cmd/tls_proxy.cpp:45:75: warning: unused parameter 'error' [-Wunused-parameter]
inline void log_error(const char* where, const boost::system::error_code& error)
                                                                          ^
./src/cmd/tls_proxy.cpp:50:44: warning: unused parameter 'where' [-Wunused-parameter]
inline void log_binary_message(const char* where, const byte buf[], size_t buf_len)
                                           ^
./src/cmd/tls_proxy.cpp:50:62: warning: unused parameter 'buf' [-Wunused-parameter]
inline void log_binary_message(const char* where, const byte buf[], size_t buf_len)
                                                             ^
./src/cmd/tls_proxy.cpp:50:76: warning: unused parameter 'buf_len' [-Wunused-parameter]
inline void log_binary_message(const char* where, const byte buf[], size_t buf_len)
                                                                           ^
./src/cmd/tls_proxy.cpp:55:35: warning: unused parameter 'where' [-Wunused-parameter]
void log_text_message(const char* where,  const byte buf[], size_t buf_len)
                                  ^
./src/cmd/tls_proxy.cpp:55:54: warning: unused parameter 'buf' [-Wunused-parameter]
void log_text_message(const char* where,  const byte buf[], size_t buf_len)
                                                     ^
./src/cmd/tls_proxy.cpp:55:68: warning: unused parameter 'buf_len' [-Wunused-parameter]
void log_text_message(const char* where,  const byte buf[], size_t buf_len)
                                                                   ^
./src/cmd/tls_proxy.cpp:272:85: warning: unused parameter 'endpoint' [-Wunused-parameter]
                       [this](boost::system::error_code ec, tcp::resolver::iterator endpoint)
                                                                                    ^
./src/cmd/tls_proxy.cpp:457:1: error: no matching constructor for initialization of
      'AppRegistrations::AppRegistration'
REGISTER_APP(tls_proxy);
^~~~~~~~~~~~~~~~~~~~~~~
./src/cmd/apps.h:71:60: note: expanded from macro 'REGISTER_APP'
#define REGISTER_APP(nm) AppRegistrations::AppRegistration g_ ## nm ## _registration(#nm, nm)
                                                           ^                         ~~~~~~~
<scratch space>:82:1: note: expanded from here
g_tls_proxy_registration
^
./src/cmd/apps.h:59:13: note: candidate constructor not viable: no known conversion from
      'int (int, char **)' to 'app_fn' (aka 'function<int (std::vector<std::string>)>') for 2nd
      argument
            AppRegistration(const std::string& name, app_fn fn)
            ^
./src/cmd/apps.h:56:13: note: candidate constructor (the implicit copy constructor) not viable:
      requires 1 argument, but 2 were provided
      class AppRegistration
            ^
./src/cmd/apps.h:56:13: note: candidate constructor (the implicit move constructor) not viable:
      requires 1 argument, but 2 were provided
10 warnings and 1 error generated.
make: *** [build/obj/app/tls_proxy.o] Error 1
$
```

--- a/src/cmd/tls_proxy.cpp
+++ b/src/cmd/tls_proxy.cpp
@@ -392,20 +392,20 @@ size_t choose_thread_count()
    return 2;
    }
 
-int tls_proxy(int argc, char* argv[])
+int tls_proxy(const std::vector<std::string> &args)
    {
-   if(argc != 6)
+   if(args.size() != 6)
       {
-      std::cout << "Usage: " << argv[0] << " listen_port target_host target_port server_cert server_key" << std::endl;
+      std::cout << "Usage: " << args[0] << " listen_port target_host target_port server_cert server_key" << std::endl;
       return 1;
       }
 
-   const size_t listen_port = to_u32bit(argv[1]);
-   const std::string target = argv[2];
-   const std::string target_port = argv[3];
+   const size_t listen_port = to_u32bit(args[1]);
+   const std::string target = args[2];
+   const std::string target_port = args[3];
 
-   const std::string server_crt = argv[4];
-   const std::string server_key = argv[5];
+   const std::string server_crt = args[4];
+   const std::string server_key = args[5];
 
    const size_t num_threads = choose_thread_count(); // make configurable
 

